src/Bicep.Core/Syntax/SyntaxExtensions.cs

```diff
@@ -20,6 +21,22 @@ public static IReadOnlyList<Diagnostic> GetParseDiagnostics(this SyntaxBase synt
 
         public static bool HasParseErrors(this SyntaxBase syntax)
             => syntax.GetParseDiagnostics().Any(d => d.Level == DiagnosticLevel.Error);
+
+        public static bool ReferencesResource(this VariableAccessSyntax syntax, ResourceDeclarationSyntax resource)
+            => StringComparer.Ordinal.Equals(syntax.Name.IdentifierName, resource.Name.IdentifierName);
+
+        public static bool NameEquals(this FunctionCallSyntax funcSyntax, string compareTo)
+            => LanguageConstants.IdentifierComparer.Equals(funcSyntax.Name.IdentifierName, compareTo);
+
+        public static bool NameEquals(this ObjectPropertySyntax propertySyntax, string compareTo)
+            => propertySyntax.Key is IdentifierSyntax idSyntax
+                && LanguageConstants.IdentifierComparer.Equals(idSyntax.IdentifierName, compareTo);
+
+        public static bool PropertyNameEquals(this PropertyAccessSyntax propAccessSyntax, string compareTo)
+                   => LanguageConstants.IdentifierComparer.Equals(propAccessSyntax.PropertyName.IdentifierName, compareTo);
+
+        public static bool IsStringLiteral(this StringSyntax stringSyntax)
+            => stringSyntax.SegmentValues.Length == 1;
```

Comment:
We have `StringSyntaxExtensions.IsInterpolated()` for this. Could you replace references to this?